io/eolearn/io/local_io.py

```diff
@@ -179,24 +180,50 @@ def execute(self, eopatch, *, filename=None):
         time_dim, height, width, band_dim = array_sub.shape
 
         index = time_dim * band_dim
-        dst_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
-        dst_crs = {'init': CRS.ogc_string(eopatch.bbox.crs)}
 
         image_dtype = array_sub.dtype if self.image_dtype is None else self.image_dtype
         if image_dtype == np.int64:
             image_dtype = np.int32
             warnings.warn('Data from feature {} cannot be exported to tiff with dtype numpy.int64. Will export as '
                           'numpy.int32 instead'.format((feature_type, feature_name)))
 
-        # Write it out to a file
-        with rasterio.open(self._get_file_path(filename, create_dir=True), 'w', driver='GTiff',
-                           width=width, height=height,
-                           count=index,
-                           dtype=image_dtype, nodata=self.no_data_value,
-                           transform=dst_transform, crs=dst_crs) as dst:
-            output_array = array_sub.astype(image_dtype)
-            output_array = np.moveaxis(output_array, -1, 1).reshape(index, height, width)
-            dst.write(output_array)
+        if not self.crs:
+            dst_crs = {'init': CRS.ogc_string(eopatch.bbox.crs)}
+            dst_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
+
+            # Write it out to a file
+            with rasterio.open(self._get_file_path(filename), 'w', driver='GTiff',
+                               width=width, height=height,
+                               count=index,
+                               dtype=image_dtype, nodata=self.no_data_value,
+                               transform=dst_transform, crs=dst_crs) as dst:
+                output_array = array_sub.astype(image_dtype)
+                output_array = np.moveaxis(output_array, -1, 1).reshape(index, width, height)
+                dst.write(output_array)
+        else:
+            src_crs = {'init': CRS.ogc_string(eopatch.bbox.crs)}
+            dst_crs = {'init': CRS.ogc_string(self.crs)}
+            src_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
+            dst_transform, width, height = rasterio.warp.calculate_default_transform(
```

Comment:
one minor comment, I would rename `width` and `height` to `dst_width` and `dst_height` to avoid confusion with original dataset. 

Similarly, on line 212 `width=dst_width` and `height=dst_height`, and on line 217 `.reshape(index, height, width)`